libraries/AP_NavEKF2/AP_NavEKF2_OptFlowFusion.cpp

```diff
@@ -56,16 +56,19 @@ void NavEKF2_core::SelectFlowFusion()
         // fuse optical flow data into the terrain estimator if available and if there is no range data (range data is better)
         fuseOptFlowData = (flowDataToFuse && !rangeDataToFuse);
         // Estimate the terrain offset (runs a one state EKF)
-        EstimateTerrainOffset();
+        if (frontend->_flowUseMask & (1<<1)) {
+            EstimateTerrainOffset();
+        }
     }
 
     // Fuse optical flow data into the main filter
-    if (flowDataToFuse && tiltOK)
-    {
+    if (flowDataToFuse && tiltOK) {
+        if (frontend->_flowUseMask & (1<<0)) {
         // Set the flow noise used by the fusion processes
         R_LOS = sq(MAX(frontend->_flowNoise, 0.05f));
         // Fuse the optical flow X and Y axis data into the main filter sequentially
```

Comment:
Will fix

Reply:
I cannot sea a situation at the moment where we would want to use optical flow data for both the main and terrain filter. This would result in a circular data dependency given of optical flow data for navigation uses the height above ground estimate and the resultant velocity estimates would then be used by the terrain estimator when it fused the flow data.  I will modify the parameter description accordingly.